### PR TITLE
Widget Based implementation of the holdfire fixes

### PR DESCRIPTION
**Pull request by ZainMGit**
Much simpler widget implementation of #6587 . Solution for units continuing to fire on enemy units when cloak/hold-fire has been enabled.

Only drops attack orders automatically gathered from "fire at will" command and does not drop manual attack commands given

Relevant threads:
https://discord.com/channels/549281623154229250/1445096052544897149
https://discord.com/channels/549281623154229250/1457499803817738362

### Work done
After holdfire or cloak command is issued ``CMD.STOP`` and ``CMD_UNIT_CANCEL_TARGET`` are issued at the start of the queue. 

#### Test steps
- [ ] Press cloak or holdfire while your commander is currently shooting at enemy units. The commander continues firing until all units are dead. 

Utilized Codex for fixing edge case

### CODE REVIEW

File: luaui/Widgets/unit_cloak_firestate.lua [hunk 1]
```diff
@@ -72,6 +75,10 @@ function widget:UnitCommand(unitID, unitDefID, teamID, cmdID, cmdParams, cmdOpts
 			if decloakFireState[unitID] ~= 0 then
 				GiveOrderToUnit(unitID, CMD.FIRE_STATE, { 0 }, 0)
 			end
+			-- Mirrors the new holdfire behavior for cloak
+			-- inserts STOP and CANCEL_TARGET at the front to break current target immediately
+			GiveOrderToUnit(unitID, CMD_INSERT, {0, CMD_STOP, 0}, {"alt"})
+			GiveOrderToUnit(unitID, CMD_INSERT, {0, CMD_UNIT_CANCEL_TARGET, 0}, {"alt"})
```
Comment (efrec) [inline luaui/Widgets/unit_cloak_firestate.lua]: Since this does "mirror" other code, do we need it?

Comment (ZainMGit) [reply luaui/Widgets/unit_cloak_firestate.lua]: In the previous commit I had used commandNotify but now that it is switched to unitcommand, i guess it is no longer explicitly needed in cloak firestate. I will remove
